config/api/src/main/java/org/jboss/windup/config/RuleSubset.java

```diff
@@ -207,9 +206,17 @@ public void perform(Rewrite rewrite, EvaluationContext context)
             Rule rule = rules.get(i);
 
             Context ruleContext = rule instanceof Context ? (Context) rule : null;
+
             long ruleTimeStarted = System.currentTimeMillis();
             try
             {
+                AbstractRuleProvider ruleProvider = (AbstractRuleProvider) ruleContext.get(RuleMetadataType.RULE_PROVIDER);
+                if (ruleProvider.getMetadata().isDisabled())
```

Comment:
I think that this needs a null check. Perhaps in the case of nested rules this can be null?